File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    vector<int> temp;
    
    int tmpsize = 0;
    for (const auto& c : score) {
        if (tmpsize < k) {
            temp.push_back(c);
            tmpsize++;
            
            sort(temp.begin(), temp.end());
        }
        else {
            if (c > temp[0]) {
                temp[0] = c;
                sort(temp.begin(), temp.end());
            }
        }
        
        answer.push_back(temp[0]);
    }
    
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    temp = []
    
    tmpsize = 0
    for i in score:
        if tmpsize < k:
            temp.append(i)
            tmpsize += 1
            temp.sort()
        else:
            if i > temp[0]:
                temp[0] = i
                temp.sort()
                
        answer.append(temp[0])
        
    return answer